### PR TITLE
Expose_the_two_last_url_child_alert_and_flood_stations

### PR DESCRIPTION
**Pull request by G-jonathan**
Expose the following URL:
http://localhost:8080/childAlert?address=<address>

This url  return a list of children (anyone aged 18 or under) living at this address.
The list include each child's first and last name, age, and a list of others
household members. If there is no child, this url  return an empty string. 

Expose the following URL:
http://localhost:8080/flood/stations?stations=<a list of station_numbers>

This url  return a list of all homes served by the barracks. This list  include the
people by address. It  also include the name, telephone number and age of the inhabitants, and
include their medical history (medications, dosage and allergies) next to each name.

### CODE REVIEW

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java [hunk 1]
```diff
@@ -184,4 +186,40 @@ public void addStationAndAddressMapping(FireStation fireStation) {
...
+        ArrayList<HouseHold> houseHoldArrayList = new ArrayList<>();
+        for (String station : stationsNumbers) {
+            ArrayList<FireStation> fireStationWhoContainsThisFireStationNumber = fireStationRepository.findFireStationByStation(station);
+            for (FireStation fireStation : fireStationWhoContainsThisFireStationNumber) {
+                ArrayList<Person> personWhoLiveAtThisAddress = personRepository.findPersonByAddress(fireStation.getAddress());
+                ArrayList<HouseHoldMember> houseHoldMemberArrayList = buildHouseHoldMemberArrayList(personWhoLiveAtThisAddress);
+                HouseHold houseHold = new HouseHold(fireStation.getAddress(), houseHoldMemberArrayList);
+                houseHoldArrayList.add(houseHold);
+            }
+
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java:202]: useless space

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java [hunk 2]
```diff
@@ -114,4 +116,38 @@ public boolean deleteAPerson(UniqueIdentifier uniqueIdentifier) {
         personInformation.setAllergies(medicalRecord.get().getAllergies());
         return Optional.of(personInformation);
     }
+
+    @Override
+    public Optional<ChildrenAndOtherMembers> getListOfChildrenLivingAtThisAddress(String address) {
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java:121]: This fun is too long, could have been great to split it in sub private functions :)

**Review by Asavarkhul — CHANGES_REQUESTED** [review]

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java (hunk 1 above)
Comment (G-jonathan) [reply src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java]: Done

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java (hunk 2 above)
Comment (G-jonathan) [reply src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/PersonServiceImpl.java:121]: Like that ?

**Review by Asavarkhul — APPROVED** [review]

**Pull request merged**